ext/phalcon/config.zep.c

```diff
@@ -409,17 +409,37 @@ PHP_METHOD(Phalcon_Config, offsetUnset) {
 PHP_METHOD(Phalcon_Config, merge) {
 
 	zend_long ZEPHIR_LAST_CALL_STATUS;
-	zval *config, config_sub;
+	zval *configParam, configParam_sub, config, _0;
 	zval *this_ptr = getThis();
 
-	ZVAL_UNDEF(&config_sub);
+	ZVAL_UNDEF(&configParam_sub);
+	ZVAL_UNDEF(&config);
+	ZVAL_UNDEF(&_0);
 
 	ZEPHIR_MM_GROW();
-	zephir_fetch_params(1, 1, 0, &config);
+	zephir_fetch_params(1, 1, 0, &configParam);
 
 
 
-	ZEPHIR_RETURN_CALL_METHOD(this_ptr, "_merge", NULL, 13, config);
+	ZEPHIR_INIT_VAR(&_0);
+	zephir_gettype(&_0, configParam TSRMLS_CC);
+	do {
+		if (ZEPHIR_IS_STRING(&_0, "array")) {
+			ZEPHIR_INIT_VAR(&config);
+			object_init_ex(&config, phalcon_config_ce);
+			ZEPHIR_CALL_METHOD(NULL, &config, "__construct", NULL, 12, configParam);
+			zephir_check_call_status();
+			break;
+		}
+		if (ZEPHIR_IS_STRING(&_0, "object")) {
+			ZEPHIR_CPY_WRT(&config, configParam);
+			break;
+		}
+		ZEPHIR_THROW_EXCEPTION_DEBUG_STR(phalcon_config_exception_ce, "Invalid data type for merge.", "phalcon/config.zep", 216);
+		return;
+	} while(0);
```

Comment:
Why wrap inside `do while` if it won't run in cycle, isn't better to just remove it?

Reply:
What about wrap `ZEPHIR_THROW_EXCEPTION_DEBUG_STR` inside else?

Reply:
It is autogenerated code  (by Zephir)

Reply:
> isn't better to just remove it?

Not better. It is sort of trick using C programming

Reply:
Yeah, I understood. But any way if elseif else will also work.

Reply:
https://github.com/php/php-src/blob/PHP-7.3.0/Zend/zend_API.h#L615

Reply:
https://stackoverflow.com/questions/1067226/c-multi-line-macro-do-while0-vs-scope-block

Reply:
OK